x-pack/plugins/rule_registry/server/utils/create_lifecycle_executor.ts

```diff
@@ -170,17 +169,21 @@ export const createLifecycleExecutor =
     > = {
       alertWithLifecycle: ({ id, fields }) => {
         currentAlerts[id] = fields;
-        return alertFactory.create(id);
+        const alert = alertFactory.create(id);
```

Comment:
this is where the rule registry gets the UUID created by the alerting framework, instead of generating it here.